samcli/lib/providers/cfn_api_provider.py

```diff
@@ -75,6 +95,208 @@ def extract_resources(self, stacks: List[Stack], collector: ApiCollector, cwd: O
+        properties = resource.get("Properties", {})
+        authorizer_type = properties.get(CfnApiProvider._AUTHORIZER_TYPE, "").lower()
+        rest_api_id = properties.get(CfnApiProvider._AUTHORIZER_REST_API)
+        name = properties.get(CfnApiProvider._AUTHORIZER_NAME)
+        authorizer_uri = properties.get(CfnApiProvider._AUTHORIZER_AUTHORIZER_URI)
+
+        if not authorizer_type:
+            raise InvalidSamTemplateException(
+                f"Authorizer '{logical_id}' is missing the '{CfnApiProvider._AUTHORIZER_TYPE}' "
+                "property, an Authorizer type must be defined."
+            )
+
+        if not rest_api_id:
+            raise InvalidSamTemplateException(
+                f"Authorizer '{logical_id}' is missing the '{CfnApiProvider._AUTHORIZER_REST_API}' "
+                "property, this must be defined."
+            )
+
+        if not name:
+            raise InvalidSamTemplateException(
+                f"Authorizer '{logical_id}' is missing the '{CfnApiProvider._AUTHORIZER_NAME}' "
+                "property, the Name must be defined."
+            )
+
+        if authorizer_type not in LambdaAuthorizer.VALID_TYPES:
+            LOG.warning(
+                "Authorizer '%s' with type '%s' is currently not supported. "
+                "Only Lambda Authorizers of type TOKEN and REQUEST are supported.",
+                logical_id,
+                authorizer_type,
+            )
+            return
+
+        if not authorizer_uri:
+            raise InvalidSamTemplateException(
+                f"Authorizer '{logical_id}' is missing the '{CfnApiProvider._AUTHORIZER_AUTHORIZER_URI}' "
+                "property, a valid Lambda ARN must be provided."
+            )
+
+        function_name = LambdaUri.get_function_name(authorizer_uri)
+        if not function_name:
+            LOG.warning(
+                "Was not able to resolve Lambda function ARN for Authorizer '%s'. "
+                "Double check the ARN format, or use more simple intrinsics.",
+                logical_id,
+            )
+            return
+
+        identity_source_template = properties.get(CfnApiProvider._AUTHORIZER_IDENTITY_SOURCE, None)
+
+        if identity_source_template is None and authorizer_type == LambdaAuthorizer.TOKEN:
+            raise InvalidSamTemplateException(
+                f"Lambda Authorizer '{logical_id}' of type TOKEN, must have "
+                f"'{CfnApiProvider._AUTHORIZER_IDENTITY_SOURCE}' of type string defined."
+            )
+
+        if identity_source_template is not None and not isinstance(identity_source_template, str):
+            raise InvalidSamTemplateException(
+                f"Lambda Authorizer '{logical_id}' contains an invalid '{CfnApiProvider._AUTHORIZER_IDENTITY_SOURCE}', "
+                "it must be a comma-separated string."
+            )
+
+        # split and parse out identity sources
+        identity_source_list = []
+
+        if identity_source_template:
```

Comment:
should there be validation on it being a comma separated list?

Reply:
Thats a good idea. I think we can further extend this to validating the actual identity source itself, likely using a set of regular expressions.

I make that change in a different PR since that also applies to the other methods of collecting authorizers, and also address the empty string comparison that is done when linking.